lib/nypl_sierra_api_client.rb

```diff
@@ -31,6 +31,23 @@ def initialize(config = {})
     @retries = 0
   end
 
+  def put (path, body, options = {})
+  options = parse_http_options options
+    # Default to POSTing JSON unless explicitly stated otherwise
+    options[:headers]['Content-Type'] = 'application/json' unless options[:headers]['Content-Type']
+
+    do_request 'put', path, options do |request|
```

Comment:
Can you clarify what's going on here? It doesn't seem like `do_request` takes a block.

Reply:
@danamansana  Honestly, I don't know what that code does. I copied it from the post request and just changed it to put.

Reply:
OK, I see it is using `yield`. Confusing, but should work. Approving.

Reply:
ok it looks like a case of bad naming? around line 131 is where the api response finally gets returned. that response is then the argument passed into that block.